packages/@stylexjs/eslint-plugin/src/utils/getPropertyPriorityAndType.js

```diff
@@ -11,14 +11,484 @@
+  'scrollMarginBlockEnd',
+  'scrollMarginInline',
+  'scrollMarginInlineStart',
+  'scrollMarginInlineEnd',
+  'scrollMarginTop',
+  'scrollMarginRight',
+  'scrollMarginBottom',
+  'scrollMarginLeft',
+  'scrollPadding',
+  'scrollPaddingBlock',
+  'scrollPaddingBlockStart',
+  'scrollPaddingBlockEnd',
+  'scrollPaddingInline',
+  'scrollPaddingInlineStart',
+  'scrollPaddingInlineEnd',
+  'scrollPaddingTop',
+  'scrollPaddingRight',
+  'scrollPaddingBottom',
+  'scrollPaddingLeft',
+  'scrollSnapType',
+  'scrollSnapAlign',
+  'scrollSnapStop',
+  'contentVisibility',
+  'containIntrinsicSize',
+  'containIntrinsicWidth',
+  'containIntrinsicHeight',
+  'containIntrinsicInlineSize',
+  'containIntrinsicBlockSize',
+  'speak',
+  'speakAs',
+
+  // positioning (starting at 58)
+  'isolation',
+  'position',
+  'zIndex',
+  'top',
+  'right',
+  'bottom',
+  'left',
+  'inset',
+  'insetBlock',
+  'insetBlockStart',
+  'insetBlockEnd',
+  'insetInline',
+  'insetInlineStart',
+  'insetInlineEnd',
+  'zoom',
+  'transformOrigin',
+  'transformBox',
+  'transformStyle',
+  'transform',
+  'translate',
+  'rotate',
+  'scale',
+  'offsetPath',
+  'offsetDistance',
+  'offsetRotate',
+
+  // layout (starting at 83)
+  'container',
+  'containerName',
+  'containerType',
+  'size',
+  'direction',
+  'unicodeBidi',
+  'float',
+  'clear',
+  'contain',
+  'overflow',
+  'overflowX',
+  'overflowY',
+  'overflowBlock',
+  'overflowInline',
+  'overflowClipMargin',
+  'overscrollBehavior',
+  'overscrollBehaviorX',
+  'overscrollBehaviorY',
+  'overscrollBehaviorInline',
+  'overscrollBehaviorBlock',
+  'display',
+  'tableLayout',
+  'borderSpacing',
+  'borderCollapse',
+  'emptyCells',
+  'columns',
+  'columnCount',
+  'columnWidth',
+  'columnFill',
+  'columnRule',
+  'columnRuleColor',
+  'columnRuleStyle',
+  'columnRuleWidth',
+  'columnSpan',
+  'widows',
+  'orphans',
+  'grid',
+  'gridArea',
+  'gridAutoColumns',
+  'gridAutoFlow',
+  'gridAutoRows',
+  'gridColumn',
+  'gridColumnEnd',
+  'gridColumnGap',
+  'gridColumnStart',
+  'gridGap',
+  'gridRow',
+  'gridRowEnd',
+  'gridRowGap',
+  'gridRowStart',
+  'gridTemplate',
+  'gridTemplateAreas',
+  'gridTemplateColumns',
+  'gridTemplateRows',
+  'flex',
+  'flexFlow',
+  'flexBasis',
+  'flexDirection',
+  'flexGrow',
+  'flexShrink',
+  'flexWrap',
+  'boxOrient',
+  'lineClamp',
+  'gap',
+  'rowGap',
+  'columnGap',
+  'placeContent',
+  'placeItems',
+  'placeSelf',
+  'alignContent',
+  'alignItems',
+  'alignSelf',
+  'justifyContent',
+  'justifyItems',
+  'justifySelf',
+  'order',
+  'breakInside',
+  'breakBefore',
+  'breakAfter',
+  'shapeOutside',
+  'shapeImageThreshold',
+  'shapeMargin',
+
+  // boxModel (starting at 165)
+  'boxSizing',
+  'aspectRatio',
+  'width',
+  'inlineSize',
+  'minWidth',
+  'maxWidth',
+  'minInlineSize',
+  'maxInlineSize',
+  'height',
+  'blockSize',
+  'minHeight',
+  'maxHeight',
+  'minBlockSize',
+  'maxBlockSize',
+  'margin',
```

Comment:
The physical properties, such as margin-left should come after the logical properties such as margin-in line-start. This is an important guarantee to maintain as that is what we will end up applied.